src/constants/categories.js

```diff
@@ -1,19 +1,115 @@
 /* @flow */
+import icons from "../icons";
+
+class Category {
+    key: string;
+    name: string;
+    byline: string;
+    icon: ReactComponent;
+
+    constructor(name:string, byline:string, icon: ReactComponent) {
+        this.key = name
+            .replace(' ', '-')
+            .replace('&', '-and-')
+            .toLowerCase();
+        this.name = name;
+        this.byline = byline;
+        this.icon = icon;
+    }
+}
 
 // TODO: Add a type (with flow) for the values of this object
 // so we can verify elsewhere that we're handling all possible outcomes
-var categories = {
-    Housing:        "housing",
-    Food:           "food",
-    "Material Aid": "material-aid",
-    Finance:        "finance",
-    Legal:          "legal",
-    Councelling:    "councelling",
-    Health:         "health",
-    "Life Skills":  "life-skills",
-    Activities:     "activities",
-    Technology:     "technology",
-    Centrelink:     "centrelink",
-    "Job Services": "job-services",
-};
+var categories:Array<Category> = [];
+
+function c(props) {
+    var {name, byline, icon} = props;
+    categories.push(new Category(name, byline, icon));
```

Comment:
Why can't we just have an array of these and have Category take an object and unpack it internally? Instead of defining macro funcs.